lib/globus_client.rb

```diff
@@ -59,70 +69,168 @@ def default_auth_url
 
   attr_accessor :config
 
+  # Send an authenticated GET request
+  # @param base_url [String] the base URL of the Globus API
+  # @param path [String] the path to the Globus API request
+  # @param params [Hash] params to get to the API
+  def get(base_url:, path:, params: {}, content_type: nil)
+    response = with_token_refresh_when_unauthorized do
```

Comment:
@jmartin-sul before this PR, there is a mix of retried GETs (in `Endpoint`) and non-retried GETs (in `Identity`). It couuuld be harmless? I'll try adding it and re-running the tests and API test.

Reply:
aaah, hmm.  i think i'd only noticed the retrying use in `Endpoint` and not the non-retried ones in `Identity`.  i guess my only worry was around losing retries for multi-step file listing operations, since an issue around that was linked in the `connection` building code that was removed from `Endpoint` (iirc from looking at it yesterday).

i'm not really sure what the right thing to do is, since we probably don't want to add retries to operations that didn't have that behavior before.  or at least not without further thought.

i'd propose that we get this PR merged, and capture the question in an issue that we could put as low to medium priority in the ready column?  i'm happy to write that if that's agreeable.  then we can figure out retry stuff separately.  at worst, if my understanding is correct, we maybe have a little flaky file listing behavior if this PR is merged and a new globus_client version released before any retriable `GET`s are added back?  that seems fine.

Reply:
ok, i filed a ticket to figure out this retry thing: https://github.com/sul-dlss/globus_client/issues/117